cryptoki/src/types/mod.rs

```diff
@@ -9,606 +9,881 @@ pub mod object;
 pub mod session;
 pub mod slot_token;
 
-use crate::{Error, Result};
+use crate::{str_from_blank_padded, Error, Result};
 use cryptoki_sys::*;
 use log::error;
 use std::convert::TryFrom;
 use std::convert::TryInto;
+use std::fmt::Formatter;
 use std::ops::Deref;
 
+trait Flags: std::fmt::Display {
```

Comment:
I think instead of a trait it could've been done with a macro instead to remove some of the code duplication, but it's alright now that it's already written out.

Reply:
I can look into that.

Reply:
Would you be opposed to me looking into this in a separate ticket/issue?  I'm definitely not well versed in writing Rust macros and it may take some time for me to do that.

Reply:
Sure, and it's not really that critical, don't even know if one or the other is actually idiomatic, just saw there's a bit of duplication, but I'm perfectly happy to just keep it as is.